### PR TITLE
feat(LayerManager): add focus trapping

### PR DESCRIPTION
**Pull request by Volodomor**

### CODE REVIEW

File: src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -89,6 +87,10 @@ export function Modal({
         onOutsideClick,
         onClose,
         contentRefs: [contentRef],
+        enableFocusTrap,
+        focusTrapContainersRefs: [contentRef],
+        focusTrapOptions,
```
Comment (ValeraS) [inline src/components/Modal/Modal.tsx:92]: I guess, these options should depend on `disableEscapeKeyDown`, `disableOutsideClick` props

File: src/components/utils/LayerManager.ts [hunk 2]
```diff
@@ -18,23 +22,39 @@ export type ContentElement =
...
+}
+
+function isNonNullable<T>(value: T): value is NonNullable<T> {
+    return value !== null && value !== undefined;
 }
 
 class LayerManager {
     private stack: LayerConfig[] = [];
     private mouseDownTarget: HTMLElement | null = null;
+    private focusTraps = new WeakMap<LayerConfig, FocusTrap>();
```
Comment (ValeraS) [inline src/components/utils/LayerManager.ts:39]: Do we need to pause/unpause focus trap ourself? From [documentation](https://github.com/focus-trap/focus-trap#one-at-a-time):
> Focus trap manages a queue of traps: if A activates; then B activates, pausing A; then C activates, pausing B; when C then deactivates, B is unpaused; and when B then deactivates, A is unpaused.

File: src/components/Modal/Modal.tsx [hunk 3]
```diff
@@ -53,6 +51,9 @@ export function Modal({
     'aria-label': ariaLabel,
     container,
     qa,
+    enableFocusTrap,
+    focusTrapOptions,
+    focusTrapRef,
```
Comment (ValeraS) [inline src/components/Modal/Modal.tsx:56]: I don't think we should give direct access to the `focus-trap` options, we need to use some abstraction here.

Comment (ValeraS) [inline src/components/Modal/Modal.tsx:56]: When do we need to control focus trap instance outside the component?

**Review by ValeraS — CHANGES_REQUESTED** [review]

**Comment by amje** [comment]
The feature was implemented in #482 and released in 4.0.0. Closing this PR

**Pull request closed without merging**